macrosynergy/management/shape_dfs.py

```diff
@@ -37,154 +43,143 @@ def reduce_df(df: pd.DataFrame, xcats: List[str] = None,  cids: List[str] = None
             filt2 = dfx['real_date'] >= pd.to_datetime(value[0])
             filt3 = dfx['real_date'] <= pd.to_datetime(value[1])
             dfx = dfx[~(filt1 & filt2 & filt3)]
-
-    xcats_in_df = dfx['xcat'].unique()
-    if xcats is None:
-        xcats = sorted(xcats_in_df)
-    else:
-        missing = sorted(set(xcats) - set(xcats_in_df))
-        if len(missing) > 0:
-            print(f'Missing cross sections: {missing}')
-            xcats.remove(missing)
```

Comment:
The printing of the information that we are missing categories are an important part of the function.